packages/block/test/header.spec.ts

```diff
@@ -290,6 +290,66 @@ tape('[Block]: Header functions', function (t) {
 
     st.end()
   })
+
+  t.test('should skip consensusFormatValidation if flag is set to false', (st) => {
+    const common = new Common({ chain: Chain.Goerli, hardfork: Hardfork.Chainstart })
+    const extraData = Buffer.concat([Buffer.alloc(1)])
+
+    try {
+      BlockHeader.fromHeaderData({ extraData }, { common, consensusFormatValidation: false })
+      st.pass(
+        'should instantiate header with invalid extraData when consensusFormatValidation === false'
+      )
+    } catch (error: any) {
+      st.fail('should not throw')
+    }
+
+    st.end()
+  })
+
+  t.test('_genericFormatValidation checks', (st) => {
+    const badHash = Buffer.alloc(31)
+
+    st.throws(
+      () => BlockHeader.fromHeaderData({ parentHash: badHash }),
+      (err: any) => err.message.includes('parentHash must be 32 bytes'),
+      'throws on invalid parent hash length'
+    )
+    st.throws(
+      () => BlockHeader.fromHeaderData({ stateRoot: badHash }),
+      (err: any) => err.message.includes('stateRoot must be 32 bytes'),
+      'throws on invalid state root hash length'
+    )
+    st.throws(
+      () => BlockHeader.fromHeaderData({ transactionsTrie: badHash }),
+      (err: any) => err.message.includes('transactionsTrie must be 32 bytes'),
+      'throws on invalid transactionsTrie root hash length'
+    )
+    let nonce = Buffer.alloc(5)
```

Comment:
Not a big deal, but I find the nonce reassignment a bit weird. Could we just either scope it differently or directly assign `fromHeaderData({nonce: Buffer.alloc(5)})` ?